test/credentials.spec.js

```diff
@@ -438,6 +438,124 @@
           creds.get();
           return validateCredentials(creds);
         });
+        it('loads via credential_process', function(done) {
```

Comment:
Nit: you can put these unit tests to a test suite(like leading with `describe('credential process')`). So than you can clear the spies easily.